sources/packages/backend/libs/test-utils/src/factories/designation-agreement.ts

```diff
@@ -7,24 +7,39 @@ import {
   User,
 } from "@sims/sims-db";
 import { INSTITUTION_TYPE_BC_PRIVATE } from "@sims/sims-db/constant";
+import { addDays, formatDate } from "@sims/utilities";
 import * as faker from "faker";
-import { createFakeInstitution } from "./institution";
-import { createFakeInstitutionLocation } from "./institution-location";
-import { createFakeUser } from "./user";
 
+/**
+ * Create a fake designation agreement.
+ * @param relations relations,
+ * - `fakeInstitution` fake institution.
+ * - `fakeInstitutionLocations` fake institution location/s.
+ * - `fakeUser` fake user
+ * @param options options,
+ * - `designationStatus` designation status, by default
+ * it is approved.
+ * @returns fake designation agreement.
+ * */
 export function createFakeDesignationAgreement(
-  fakeInstitution?: Institution,
-  fakeUser?: User,
-  fakeInstitutionLocations?: InstitutionLocation[],
-  designationStatus = DesignationAgreementStatus.Approved,
+  relations: {
+    fakeInstitution: Institution;
+    fakeInstitutionLocations: InstitutionLocation[];
+    fakeUser: User;
+  },
+  options?: {
```

Comment:
Should we change this to use the below pattern to allow other possible initialization values in the future?
```ts
options?: {
  initialValue?: Partial<DesignationAgreement>;
},
```